tests/io/v3/was/scans/test_scans_api.py

```diff
@@ -119,14 +123,197 @@ def test_launch(api):
+            'config_id': '988cd296-58b0-419e-bd43-0d884080daf6',
+            'asset_id': 'f8598420-1e33-4aed-abd0-56d114e96d3e',
+            'target': 'http://example.com/',
+            'created_at': '2020-12-07T20:00:14.319827Z',
+            'updated_at': '2020-12-07T20:01:19.815217Z',
+            'started_at': '2020-12-07T20:01:00.386897Z',
+            'finalized_at': '2020-12-07T20:01:19.808199Z',
+            'requested_action': 'start',
+            'status': 'completed',
+            'metadata': {
+                'found_urls': 1,
+                'queued_urls': 0,
+                'scan_status': 'running',
+                'audited_urls': 1,
+                'queued_pages': 0,
+                'audited_pages': 1,
+                'request_count': 163,
+                'response_time': 0
+            },
+            'scanner': {
+                'group_name': 'US Cloud Scanner'
+            },
+            'template_name': 'config_audit'
+        },
+            {
+                'scan_id': '6f1ed550-7894-40b3-a6ea-3be06f9338a4',
+                'user_id': '01c924ee-e6ab-4d55-b283-dffba6dcce4c',
+                'config_id': '988cd296-58b0-419e-bd43-0d884080daf6',
+                'asset_id': 'f8598420-1e33-4aed-abd0-56d114e96d3e',
+                'target': 'http://example.com/',
+                'created_at': '2020-12-04T20:00:20.058359Z',
+                'updated_at': '2020-12-04T20:01:34.843822Z',
+                'started_at': '2020-12-04T20:01:15.837083Z',
+                'finalized_at': '2020-12-04T20:01:34.831731Z',
+                'requested_action': 'start',
+                'status': 'completed',
+                'metadata': {
+                    'found_urls': 1,
+                    'queued_urls': 0,
+                    'scan_status': 'running',
+                    'audited_urls': 1,
+                    'queued_pages': 0,
+                    'audited_pages': 1,
+                    'request_count': 162,
+                    'response_time': 0
+                },
+                'scanner': {
+                    'group_name': 'US Cloud Scanner'
+                },
+                'template_name': 'config_audit'
+        }
+        ]
+    }
+    QP = '?limit=200&sort=name&sort=order&offset=0'
+    responses.add(
+        responses.POST,
+        f'{WAS_SCANS_BASE_URL}/configs/{SAMPLE_CONFIG_ID}/scans/search{QP}',
+        match=[matchers.json_params_matcher(payload)],
+        json=api_response
+    )
+
+    iterator = api.v3.was.scans.search(SAMPLE_CONFIG_ID,
+                                       fields=fields,
+                                       limit=200,
+                                       sort=sort,
+                                       filter=filters)
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_response['pagination']['total']
```

Comment:
Refactor above code, to get rid of loop and rewrite it in pythonic way.

Reply:
removed for loop here and used `len(list(iterator))` in the assert statement